src/pages/home/report/ReportActionItemFragment.js

```diff
@@ -94,70 +90,45 @@ const defaultProps = {
     isThreadParentMessage: false,
     isApprovedOrSubmittedReportAction: false,
     isFragmentContainingDisplayName: false,
-    displayAsGroup: false,
 };
 
 function ReportActionItemFragment(props) {
-    switch (props.fragment.type) {
+    const fragment = props.fragment;
+
+    switch (fragment.type) {
         case 'COMMENT': {
-            const {html, text} = props.fragment;
-            const isPendingDelete = props.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE && props.network.isOffline;
+            const isPendingDelete = props.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE;
 
             // Threaded messages display "[Deleted message]" instead of being hidden altogether.
             // While offline we display the previous message with a strikethrough style. Once online we want to
             // immediately display "[Deleted message]" while the delete action is pending.
 
-            if ((!props.network.isOffline && props.isThreadParentMessage && props.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) || props.fragment.isDeletedParentAction) {
+            if ((!props.network.isOffline && props.isThreadParentMessage && isPendingDelete) || props.fragment.isDeletedParentAction) {
                 return <RenderHTML html={`<comment>${props.translate('parentReportAction.deletedMessage')}</comment>`} />;
             }
 
-            // If the only difference between fragment.text and fragment.html is <br /> tags
-            // we render it as text, not as html.
-            // This is done to render emojis with line breaks between them as text.
-            const differByLineBreaksOnly = Str.replaceAll(html, '<br />', '\n') === text;
-
-            // Only render HTML if we have html in the fragment
-            if (!differByLineBreaksOnly) {
-                const editedTag = props.fragment.isEdited ? `<edited ${isPendingDelete ? 'deleted' : ''}></edited>` : '';
-                const htmlContent = isPendingDelete ? `<del>${html}</del>` : html;
-
-                const htmlWithTag = editedTag ? `${htmlContent}${editedTag}` : htmlContent;
+            // Does the fragment content represent an attachment?
+            const isFragmentAttachment = ReportUtils.isReportMessageAttachment(fragment);
 
-                return <RenderHTML html={props.source === 'email' ? `<email-comment>${htmlWithTag}</email-comment>` : `<comment>${htmlWithTag}</comment>`} />;
+            if (isFragmentAttachment) {
+                return (
+                    <AttachmentCommentFragment
```

Comment:
I created it for symmetry. I agree that it's not _strictly_ needed, but do you think it needs to be removed now? It can grow if New Expensify goes more into a fat client architecture direction.

Reply:
If it was me reviewing the proposal I would _personally_ say to not bother with this so we can move forward.

I did find this:

https://github.com/Expensify/App/blame/aeb7f28091bdd8c0fffa7ed5158e41d2227d06a9/contributingGuides/REVIEWER_CHECKLIST.md#L44

But that doesn't really instruct us on _when_ it's appropriate to create a new component.

Our style guide doesn't say much on this so the ideal thing would be to hash out what the best practice should be and then pop the outcome in the style guide so people can put their bias aside.

Reply:
I'm not much of a checklist/style guide person (I appreciate automated code formatters, though!), so let me know what can we do to resolve this thread in the context of this PR

If we were forced to add a new checkbox or something similar, I would vote for:

> If there's nothing wrong with a chunk of PR code being reviewed, and there's no confidence that any alternative would be apparently or measurably better, it should be left as-is